caramel/config.py

```diff
@@ -37,3 +37,37 @@ def __call__(self, parser, namespace, values, option_string=None):
                 " specify a path in the call like so:\n"
                 "\t caramel_initializedb /path/to/config.ini [...]"
             )
+
+
+def _get_config_value(
+    arguments: argparse.Namespace,
+    variable,
+    required=False,
+    setting_name=None,
+    settings=None,
+    env=None,
+):
+    """Returns what value to use and logs where it got it from, prefer argument >
+    env-variable > config-file"""
+    if env is None:
+        env = os.environ
+    env_var = "CARAMEL_" + variable.upper().replace("-", "_")
+
+    if setting_name is None:
+        setting_name = variable
+
+    config_value = None
+    if settings is not None:
+        config_value = settings.get(setting_name, config_value)
+
+    config_value = env.get(env_var, config_value)
+
+    arg_value = getattr(arguments, variable, config_value)
+    config_value = arg_value if arg_value is not None else config_value
+
+    if required and config_value is None:
+        raise ValueError(
+            "No " + variable + " could be found as either an argument,"
```

Comment:
This should be an f-string:

f"No {variable} could be found as either an argument, in the environemnt variable {env_var} or in the config file"

--- 
 ( below here is just commentary)

Also, since Exceptions can pack extra data, you can _also_ attach the variables to the exception  Like

"ValueError(msg, variable, envvar)"   if you wish to pass the data out of the function, if you fex. want to handle things later and know them.